src/modules/dashboard/components/TaskList/TaskListItem.tsx

```diff
@@ -76,40 +79,46 @@ const TaskListItem = ({ data, intl: { formatMessage } }: Props) => {
     );
   }
 
+  const handleClick = () => {
+    history.push({
+      pathname: `/colony/${colonyName}/task/${draftId}`,
+    });
+  };
+
   return (
-    <Link
-      className={styles.globalLink}
-      to={`/colony/${colonyName}/task/${draftId}`}
-    >
-      <TableRow>
-        <TableCell className={styles.taskDetails}>
-          <p className={styles.taskDetailsTitle}>{title || defaultTitle}</p>
-          {reputation ? (
-            <span className={styles.taskDetailsReputation}>
-              <FormattedMessage
-                {...MSG.reputation}
-                values={{ reputation: reputation.toString() }}
-              />
-            </span>
-          ) : null}
-        </TableCell>
-        <TableCell className={styles.taskPayouts}>
-          {!!availableTokens && (
-            <PayoutsList
-              payouts={payouts}
-              nativeToken={nativeTokenRef}
-              tokenOptions={availableTokens}
+    <TableRow className={styles.globalLink} onClick={() => handleClick()}>
+      <TableCell className={styles.taskDetails}>
+        <p className={styles.taskDetailsTitle}>{title || defaultTitle}</p>
+        {reputation ? (
+          <span className={styles.taskDetailsReputation}>
+            <FormattedMessage
+              {...MSG.reputation}
+              values={{ reputation: reputation.toString() }}
             />
-          )}
-        </TableCell>
-        <TableCell className={styles.userAvatar}>
-          {workerAddress && <UserAvatar size="s" address={workerAddress} />}
-        </TableCell>
-      </TableRow>
-    </Link>
+          </span>
+        ) : null}
+      </TableCell>
+      <TableCell className={styles.taskPayouts}>
+        {!!availableTokens && (
+          <PayoutsList
+            payouts={payouts}
+            nativeToken={nativeTokenRef}
+            tokenOptions={availableTokens}
+          />
+        )}
+      </TableCell>
+      <TableCell className={styles.userAvatar}>
+        {workerAddress && <UserAvatar size="s" address={workerAddress} />}
+      </TableCell>
+    </TableRow>
   );
 };
 
 TaskListItem.displayName = displayName;
 
-export default injectIntl(TaskListItem);
+const enhance = compose(
+  withRouter,
+  injectIntl,
+) as any;
```

Comment:
I don't know if that _can_ be solved properly right now but it definitely is a code smell. We introduced a lot of `as any`s during the typescript refactor but that doesn't mean we can add more now. If any we should try to remove as many of those as possible. What's the typescript error if you remove it? Let's find a better solution here.

Reply:
Yeih, found a type, after trying for a while, I will add it in, in other occurrences as well.